consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/MessageValidator.java

```diff
@@ -17,20 +17,63 @@
 import org.hyperledger.besu.consensus.qbft.messagewrappers.Commit;
 import org.hyperledger.besu.consensus.qbft.messagewrappers.Prepare;
 import org.hyperledger.besu.consensus.qbft.messagewrappers.Proposal;
+import org.hyperledger.besu.ethereum.core.Hash;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class MessageValidator {
 
-  public MessageValidator() {}
+  private static final Logger LOG = LogManager.getLogger();
+
+  @FunctionalInterface
+  public interface PrepareValidatorFactory {
+    PrepareValidator create(final Hash expectedDigest);
```

Comment:
nit: final on the interface isn't necessary or useful

Reply:
done